docs/How-to-implement-a-new-ci-test.md

```diff
@@ -0,0 +1,146 @@
+    For example: Can be used to check if the given level is below the max level.
+
+
+You will use your Kernel in the next step.
+
+
+## Implement your own skeleton discovery
+
+Following is a template you can use for the skeleton discovery. Just copy and adapt it tho your special use case. 
+
+```
+@timed
+def discover_skeleton_gpu_discrete(
+    skeleton: np.ndarray,
+    data: np.ndarray,
+    alpha: float,
+    max_level: int,
+    num_variables: int,
+    num_observations: int,
+    kernels=None,
+    is_debug: bool = False,
+    should_log: bool = False,
+) -> SkeletonResult:
+
+    # initialize the data structures you need on the gpu using cupy and do your necessary setup
+
+     d_skeleton = cp.asarray(skeleton, dtype=cp.uint16)
+     d_seperation_sets = cp.full(num_variables * num_variables * max_level, -1, np.int32)
+     d_pmax = cp.full((num_variables, num_variables), 0, np.float32)
+
+    if kernels == None:
+        # kernel in the singular here as in our own CI tests we always used two kernels -> plural
+        # here we only assume one kernel, but you can of course use as many as you like
+        kernel = YourKernel(
+            your_needed_params, 
+            is_debug=is_debug,
+            should_log=should_log
+        )
+
+    # init stream we will need later on for synchronization
+    stream = cp.cuda.get_current_stream()
+
+    kernel_start = timer()
+
+    for level in range(0, max_level + 1):
+        # prepare whatever you need and maybe check if you are already finished with the pc algorithm
+        # then execute the actual kernel, by using the kernel management you implemented earlier
+        kernel(level, kernel_params) 
+       
+
+    # synchronize the stream so the times measurement is accurate and not just done later
+    stream.synchronize()
+    kernel_time = timer() - kernel_start
+
+    # fetch the skeleton from the gpu
+    result_skeleton = d_skeleton.get()
+
+    # fetch separation_sets and bring them in assumed form for edge orientation
+    separation_sets = d_seperation_sets.get().reshape(
+        (num_variables, num_variables, max_level)
+    )
+
+    # bring the pmax values in the same form pcalg would return
+    pmax = postprocess_pmax(d_pmax.get())
+
+    return SkeletonResult(result_skeleton, separation_sets, pmax, kernel_time)
+```
+
+Note: make sure the types you initialize your data with on python side matches the data types the CUDA kernel takes. An example in the template is d_skeleton that is initialized as cp.uint16.  
+
+After implementing the kernel discovery you can extend the pc function (gpucsl/pc/pc.py). First add your distribution to the DataDistribution enum in the same file. Then in pc test for your DataDistribution and call your skeleton discovery like it is done for DataDistribution.GAUSSIAN or DataDistribution.DISCRETE. Depending on whether you need more arguments you have to extend the pc functions argument list. 
```

Comment:
```suggestion
After implementing the kernel discovery you can extend the pc function (gpucsl/pc/pc.py). First, add your distribution to the DataDistribution enum in the same file. Then in pc test for your DataDistribution and call your skeleton discovery like it is done for DataDistribution.GAUSSIAN or DataDistribution.DISCRETE. Depending on whether you need more arguments you have to extend the pc functions argument list. 
```
"Then in pc test for your DataDistribution and call your skeleton discovery like..." sounds a little bit strange for me. Maybe, some words got lost? =)

Reply:
Nope. Just my shitty english🙈. Will get changed in the other PR anyways as this part changed -> not worth changing it